File: agents/csharp/NodeTraceAgent/Services/TokenService.cs
using System.Text.Json;

namespace NodeTraceAgent.Services
{
    public record TokenData(string Token, string DeviceId);

    public class TokenService
    {
        private readonly string _path = "token.json";

        public void SaveToken(string token, string deviceId)
        {
            var data = new { Token = token, DeviceId = deviceId };
            var json = JsonSerializer.Serialize(data);
            File.WriteAllText(_path, json);
        }

        public (string? Token, string? DeviceId) LoadToken()
        {
            if (!File.Exists(_path))
                return (null, null);

            var json = File.ReadAllText(_path);
            var data = JsonSerializer.Deserialize<TokenData>(json);
            return (data?.Token, data?.DeviceId);
        }
    }
}

File: agents/python/services/token_service.py
import json
import os

class TokenService:
    FILE = "token.json"

    def save(self, token, device_id):
        with open(self.FILE, "w") as f:
            json.dump({"token": token, "device_id": device_id}, f)

    def load(self):
        if not os.path.exists(self.FILE):
            return None, None
        with open(self.FILE, "r") as f:
            data = json.load(f)
            return data.get("token"), data.get("device_id")